workflow-bot-app/src/test/java/com/symphony/bdk/workflow/RoomIntegrationTest.java

```diff
@@ -241,6 +243,37 @@ void addRoomMember() throws Exception {
     verify(streamService, timeout(5000)).addMemberToRoom(456L, "abc");
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {"/room/obo/add-room-member-obo-valid-username.swadl.yaml",
+      "/room/obo/add-room-member-obo-valid-userid.swadl.yaml"})
+  void addRoomMemberObo(String workflowFile) throws Exception {
+    final Workflow workflow =
+        SwadlParser.fromYaml(getClass().getResourceAsStream(workflowFile));
+
+    when(bdkGateway.obo(any(String.class))).thenReturn(botSession);
+    when(bdkGateway.obo(any(Long.class))).thenReturn(botSession);
+
+    engine.deploy(workflow);
+    engine.onEvent(messageReceived("/add-room-member"));
```

Comment:
maybe we should run this line twice to make sure the cache is working, the session is reused.

Reply:
"bdkGateway" object that uses the cache, is a mock. We cannot really test that here.